hamlpy/parser/attributes.py

```diff
@@ -144,11 +130,17 @@ def read_attribute_dict(stream):
     """
     data = OrderedDict()
 
-    start, terminator = stream.text[0], stream.text[-1]
```

Comment:
Previously this code assumed the stream only contained an attribute dictionary - now it contains the entire haml document